import sys
from PyQt5.QtGui import QIcon
from PyQt5.QtWidgets import QApplication, QMainWindow
from PyQt5.uic import loadUi

import requests
from bs4 import BeautifulSoup
from fake_useragent import UserAgent


class MyApp(QMainWindow):
    url = 'https://www.tgju.org/'

    def __init__(self):
        super().__init__()
        loadUi("UI/main_window2.ui", self)  # Ù…Ø³ÛŒØ± ÙØ§ÛŒÙ„ .ui
        self.setWindowTitle("Chande - Ú†Ù†Ø¯ÙØŸ")
        self.setWindowIcon(QIcon("logo/logochande.png"))

        self.pushButton.clicked.connect(self.refresh)

    def refresh(self):
        try:
            # Ø³Ø§Ø®ØªÙ† User-Agent Ø¬Ø¹Ù„ÛŒ
            user_agent = UserAgent()
            ua = user_agent.random

            # Ø§Ø±Ø³Ø§Ù„ Ø¯Ø±Ø®ÙˆØ§Ø³Øª Ø¨Ù‡ Ø³Ø§ÛŒØª
            self.label_21.setText('Ø¨Ø±ÙˆØ² Ø±Ø³Ø§Ù†ÛŒ...')
            QApplication.processEvents()
            response = requests.get(self.url, headers={'User-Agent': ua})
            if response.status_code != 200:
                self.label_21.setText("âŒ Ø§ØªØµØ§Ù„ Ø¨Ø±Ù‚Ø±Ø§Ø± Ù†Ø´Ø¯")
                self.label_9.setText("-")
                self.label_10.setText("-")
                return

            # Ù¾Ø±Ø¯Ø§Ø²Ø´ HTML
            self.label_21.setText("Ø§Ù†Ù„Ø§ÛŒÙ† ğŸŸ¢ğŸŸ¢")
            soup = BeautifulSoup(response.text, 'html.parser')

            # Ù¾ÛŒØ¯Ø§ Ú©Ø±Ø¯Ù† Ù‚ÛŒÙ…Øªâ€ŒÙ‡Ø§ Ø¨Ø± Ø§Ø³Ø§Ø³ Ú©Ù„Ø§Ø³
            market_price = soup.find('td', class_='market-price')
            market_high = soup.find('td', class_='market-high')
            market_low = soup.find('td', class_='market-low')

            # Ø¨Ø±Ø±Ø³ÛŒ ÙˆØ¬ÙˆØ¯ ØªÚ¯â€ŒÙ‡Ø§
            if market_price and market_high and market_low:
                self.label_8.setText(f"ğŸ’° {market_price.text.strip()}")
                self.label_9.setText(f"ğŸ“ˆ{market_high.text.strip()}")
                self.label_10.setText(f"ğŸ“‰{market_low.text.strip()}")
            else:
                self.label_8.setText("â— Ø§Ø·Ù„Ø§Ø¹Ø§Øª ÛŒØ§ÙØª Ù†Ø´Ø¯")
                self.label_9.setText("-")
                self.label_10.setText("-")

        except Exception as e:
            self.label_8.setText("âš ï¸ Ø®Ø·Ø§ÛŒ Ø¯Ø§Ø®Ù„ÛŒ")
            self.label_9.setText(str(e))
            self.label_10.setText("-")


if __name__ == "__main__":
    app = QApplication(sys.argv)
    window = MyApp()
    window.show()
    sys.exit(app.exec_())
